### PR TITLE
make safe calls to window

### PR DESCRIPTION
**Pull request by SubhrajyotiSen**

### CODE REVIEW

**Comment by pwittchen** [comment]
Are these safe calls needed here? What's the reason behind adding them?

**Comment by SubhrajyotiSen** [comment]
window is tagged with @Nullable

File: CameraXBasic/app/src/main/java/com/android/example/cameraxbasic/utils/ViewExtensions.kt [hunk 1]
```diff
@@ -75,16 +75,16 @@ fun View.padWithDisplayCutout() {
 /** Same as [AlertDialog.show] but setting immersive mode in the dialog's window */
 fun AlertDialog.showImmersive() {
     // Set the dialog to not focusable
-    window.setFlags(
+    window?.setFlags(
```
Comment (piotrek1543) [inline CameraXBasic/app/src/main/java/com/android/example/cameraxbasic/utils/ViewExtensions.kt]: @SubhrajyotiSen I think that you could simplify code using `nullable.let {   }` to to remove doubled `window?...`

Comment (SubhrajyotiSen) [reply CameraXBasic/app/src/main/java/com/android/example/cameraxbasic/utils/ViewExtensions.kt]: Good point

**Comment by owahltinez** [comment]
Thanks for this PR @SubhrajyotiSen. The version that you had before using `let` was actually better, since if `window` is null (for whatever strange reason) then `show()` is never called.

**Comment by SubhrajyotiSen** [comment]
@owahltinez So should I put the `show()` call outside the `let {}` block?

**Comment by owahltinez** [comment]
I think that we should change all calls from `window` to `window?` in the original code instead.

**Comment by piotrek1543** [comment]
@owahltinez wait, I'm pretty sure that `variable?.let { //... }` is equal to `if (variable != null) { //... }`  so it would be called once and then taking operations on non null object.

**Comment by SubhrajyotiSen** [comment]
I feel it the calls on `window` should be wrapped in `let {}` and the `show()` call should be outside

**Comment by owahltinez** [comment]
If you want to use `let` and keep `show` outside, then you'll need to have two calls to `let` to maintain the current order of operations. Personally, I think it would be better to just replace `window` with `window?` in the original code.

**Comment by SubhrajyotiSen** [comment]
@owahltinez That makes sense now. Should I update the PR?

**Comment by SubhrajyotiSen** [comment]
@owahltinez That makes sense now. Should I update the PR?

**Comment by owahltinez** [comment]
Yes, please. I think it should look like in https://github.com/SubhrajyotiSen/camera/commit/6359b03175e0d7980b5ad7fe2a1906775f512cbe.

**Comment by owahltinez** [comment]
Sorry for the delay, I have been trying to figure out the process for merging pull requests. Looks good to me, merging!

**Pull request merged**